src/Driver/Pdo/AbstractPdoCommand.php

```diff
@@ -255,21 +255,13 @@ protected function internalGetQueryResult(int $queryMode): mixed
         return $result;
     }
 
-    /**
-     * Logs the current database query if query logging is on and returns the profiling token if profiling is on.
-     */
-    protected function logQuery(string $rawSql, string $category): void
-    {
-        $this->logger?->log(LogLevel::INFO, $rawSql, [$category]);
-    }
-
     protected function queryInternal(int $queryMode): mixed
     {
         $logCategory = self::class . '::' . $this->getQueryMode($queryMode);
+        $rawSql = $this->getRawSql();
```

Comment:
Do not call `CommandInterface::getRawSql()` if no logger or profiler

See https://github.com/yiisoft/db/pull/781